secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/common/jwt/SapiJwt.java

```diff
@@ -0,0 +1,120 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.common.jwt;
+
+import org.forgerock.json.jose.jws.SignedJwt;
+import org.forgerock.util.Reject;
+
+/**
+ * Base class that can be used to access common data held within a Json Web Token (JWT). This base class expects a
+ * Jwt to have the following:
+ * - a b64 url encoded string representation
+ * - a SignedJwt representation
+ * - a ClaimSetFacade allowing access to the Jwt's underlying Claims
+ * - a kid claim (key Id) in the header
+ * - an issuer claim
+ */
+public class SapiJwt {
+
+    private final String b64EncodedJwtString;
+    private final SignedJwt signedJwt;
+    private final ClaimsSetFacade claimsSet;
+    private final String keyId;
+    private final String issuer;
+    private boolean signatureHasBeenValidated;
+
+    public SapiJwt(SapiJwtBuilder builder) {
+        this.b64EncodedJwtString = builder.getB64EncodedJwtString();
+        this.signedJwt = builder.getSignedJwt();
+        this.claimsSet = builder.getClaimsSet();
+        this.keyId = builder.getKid();
+        this.issuer = builder.getIssuer();
+        this.signatureHasBeenValidated = false;
+        Reject.ifNull(issuer, "issuer must not be null");
+        Reject.ifNull(keyId, "kid must not be null");
+        Reject.ifNull(b64EncodedJwtString, "b64EncodedJwtString must not be null");
+        Reject.ifNull(signedJwt, "signedJwt must not be null");
+        Reject.ifNull(claimsSet, "claimsSet must not be null");
+    }
+
+    /**
+     * Indicates if the data in the Software Statement can be trusted, i.e. if the signature has been
+     * validated against the relevant JWKS.
+     * @return true if the software statement signature has been validated, false if it has not. Note that if this
+     * method returns false it does not mean that the signature is invalid, mearly that it has not been validated. It
+     * may be valid, but not have been validated.
+     */
+    public boolean signatureHasBeenValidated() {
+        return signatureHasBeenValidated;
+    }
+
+    /**
+     * Set a flag that indicates if the data in the Software Statement can be trusted, i.e. if the signature has been
+     * validated against the relevant JWKS
+     * @param signatureHasBeenValidated true if the signature has been validated, false if it has not.
+     */
+    public void setSignatureHasBeenValidated(boolean signatureHasBeenValidated) {
+        this.signatureHasBeenValidated = signatureHasBeenValidated;
+    }
```

Comment:
It would be better to manage this state inside this class, otherwise it is open to abuse as anyone can set the flag to true.

The nimbus lib does this, see classes: com.nimbusds.jwt.SignedJWT  / com.nimbusds.jose.JWSObject

Happy for this to be merged as is and fixed in another issue